py3status/core.py

```diff
@@ -677,6 +678,21 @@ def setup(self):
             # load and spawn i3status.conf configured modules threads
             self.load_modules(self.py3_modules, user_modules)
 
+        # determine the target output format (i3bar or tmux)
+        self.output_format = self.config["py3_config"]["general"]["output_format"]
+
+        # tmux allows configurable separators between bar entries.
+        # handle configured value or fall back to the matching default for output_format
+        self.output_format_separator = self.config["py3_config"]["py3status"].get(
+            "output_format_separator", DEFAULT_SEPARATORS[self.output_format]
```

Comment:
There could be others like `dzen2`. https://i3wm.org/docs/i3status.html#_general 
Edge cases to be tested.

Reply:
thanks for the link. this hints at the larger question whether py3status should support all values of output_format that i3status supports, although it's not immediately obvious to me how i3status integrates into py3status when the output_format is not i3bar. In my testing, I have so far actually not used any i3stauts modules, only pure py3status.

It might make sense to retire this PR for now, and carefully review the interaction between i3status and py3status with regards to different values of output_status before proceeding here.

Also, I see that i3status has a `separator` config option that covers the use case for which this PR introduced output_format_separator. Another good reason to review.

Reply:
initial testing suggests that in fact the i3status integration of py3status will not work when output_format is anything else than "i3bar". setting a value of "tmux" will cause i3status to crash and py3status to emit a warning, setting a value of "term" will cause py3status to hang in setup.

Reply:
>not immediately obvious to me how i3status integrates into py3status...

Since the inception, py3status was a python wrapper that could read i3status config and still be enhanced more with python (be it a modules or whatever).

Nowadays, I think it could be easier to break up i3status into its own module similar to `i3block`, `i3pystatus` and maybe `conky`... meaning for every i3status module you run, the process get spawned instead of currently one process for as many i3status modules as you want to add.

Big task to tackle on.... as we're not 100% there with i3status, but is more than enough for most users... and we would have to mimic any new or missing i3status features too.